src/cargo/util/command_prelude.rs

```diff
@@ -742,6 +743,10 @@ Run `{cmd}` to see possible targets."
             gctx.cli_unstable()
                 .fail_if_stable_opt("--unit-graph", 8002)?;
         }
+        if build_config.checksum_freshness {
+            gctx.cli_unstable()
+                .fail_if_stable_opt("--checksum-freshness", 14136)?;
+        }
```

Comment:
If we're defining a feature flag for this, we don't need to require `-Zunstable-options` (which means we should also stop having the tests set it)

Reply:
Resolved in 0240bf05d640eae07d6d74523e14eea303db983a